packages/eslint-plugin/src/rules/restrict-template-expressions.ts

```diff
@@ -76,6 +78,13 @@ export default util.createRule<Options, MessageId>({
         return true;
       }
 
+      if (
+        options.allowRegExp &&
+        util.getTypeName(typeChecker, type) === 'RegExp'
```

Comment:
Why not `util.isTypeFlagSet`?

Reply:
https://github.com/microsoft/TypeScript/blob/main/src/compiler/types.ts#L5048 doesn't contain flags for `RegExp`.